facebook/react-native
# [CLI] Copy assets other then fonts

This PR allows assets other then font files to be copied & linked into project using `react-native link` command.

For now `react-native link` only allows certain asset files (font file for android, font + image for ios) to be copied/linked, but it's sometimes useful for an external RN library to do so. For example, a 3D library may want to load several .stl/.obj modeling file, or a sound effect library may want to load some audio samples.

As a RN library developer, currently we can only write manual installation step in README to guide our users, however it's tedious and often confuse people if they have no native developing experience. 

## Test Plan

Precondition: we are in the root of some RN project

1. create a fake library in node_modules (in here we call it `rn-test-asset`), with following structure:
![image](https://user-images.githubusercontent.com/10307875/41028448-0b369a1a-69ac-11e8-89f2-0784655d4d10.png)

(we can put arbitrary content in `test.txt`, it doesn't matter)

2. the `package.json` for this fake library should contains an `assets` config, like this:
![image](https://user-images.githubusercontent.com/10307875/41028756-b0bcc16c-69ac-11e8-921d-54520f9ef6b7.png)

3. run `react-native link rn-test-asset`

4. verify whether the asset files (for now, it should be `test.txt`) has been linked:

* for iOS, we should check `Resources` and `Copy Bundle Resources`:
![image](https://user-images.githubusercontent.com/10307875/41029036-3f95611e-69ad-11e8-820d-869942bba417.png)
![image](https://user-images.githubusercontent.com/10307875/41029154-8bed9ab8-69ad-11e8-99db-66e399311fad.png)

* for android, we should check `android/app/src/main/assets/`:
![image](https://user-images.githubusercontent.com/10307875/41029279-deb160a4-69ad-11e8-8b00-3063be4a0c1d.png)

5. run `react-native unlink rn-test-asset`

6. verify the asset is properly unlinked

## Related PRs

None

## Release Notes

[CLI][ENHANCEMENT][local-cli/link/{ios,android}/{copyAssets,unlinkAssets}.js] - Allow react-native link to link asset other then font files
